web/base_viewer.js

```diff
@@ -262,6 +265,18 @@ class BaseViewer {
     this.maxCanvasPixels = options.maxCanvasPixels;
     this.l10n = options.l10n || NullL10n;
     this.#enablePermissions = options.enablePermissions || false;
+    this.pageColors = options.pageColors || null;
+
+    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
+      if (
+        options.pageColors &&
+        options.pageColors.foreground === "CanvasText" &&
+        options.pageColors.background === "Canvas" &&
```

Comment:
Isn't this too specific to be *generally* helpful though?

What if the user e.g. flips the values of background/foreground, which I don't believe would be caught by these checks?